projects/storefrontapp-e2e-cypress/cypress/helpers/infinite-scroll.ts

```diff
@@ -0,0 +1,181 @@
+}
+
+export function backToTopIsVisisble(isShowMoreButton?: boolean) {
+  if (isShowMoreButton) {
+    cy.get(`.cx-${doubleButton}-btn-container`).should('be.visible');
+  } else {
+    cy.get(`.cx-${singleButton}-btn-container`).should('be.visible');
+  }
+}
+
+export function backtoTopIsNotVisible() {
+  cy.get(`.cx-${singleButton}-btn-container`).should('not.be.visible');
+}
+
+export function scrollToFooter(
+  isShowMoreButton?: boolean,
+  productLimit?: number
+) {
+  if (productLimit) {
+    defaultProductLimit = productLimit;
+  }
+
+  cy.get('cx-breadcrumb h1').then($breadcrumb => {
+    const breadcrumbQuantityNumber = Number($breadcrumb.text().split(' ')[0]);
+
+    numberOfIteration = Math.floor(
+      breadcrumbQuantityNumber / defaultProductLimit
+    );
+
+    let numberOfProducts = defaultNumberOfProducts;
+
+    for (let i = 1; i < numberOfIteration; i++) {
+      if (isShowMoreButton) {
+        cy.get('div')
+          .contains('SHOW MORE')
+          .click({ force: true })
+          .wait(`@${productLoadedQuery}`)
+          .then(() => {
+            numberOfProducts += defaultNumberOfProducts;
+            cy.get('cx-product-list-item').should(
+              'have.length',
+              numberOfProducts
+            );
+          });
+      } else {
+        cy.scrollTo('bottom', { easing: 'linear', duration: scrollDuration })
+          .wait(`@${productLoadedQuery}`)
+          .then(() => {
+            numberOfProducts += defaultNumberOfProducts;
+            cy.get('cx-product-list-item').should(
+              'have.length',
+              numberOfProducts
+            );
+          });
+      }
+    }
+  });
+}
+
+export function verifySortingResetsList() {
+  cy.get('cx-sorting .ng-select:first').ngSelect(
+    PRODUCT_LISTING.SORTING_TYPES.BY_TOP_RATED
+  );
+
+  cy.wait(`@${productLoadedQuery}`);
+
+  assertDefaultNumberOfProducts('list');
+}
+
+export function verifyFilterResetsList() {
+  clickFacet('Brand');
+
+  cy.wait(`@${productLoadedQuery}`);
+
+  assertDefaultNumberOfProducts('list');
+}
+
+export function verifyGridResetsList() {
+  cy.get('cx-product-view > div > div:first').click({ force: true });
+
+  cy.wait(`@${productLoadedQuery}`);
+
+  assertDefaultNumberOfProducts('grid');
+}
+
+export function infiniteScrollNoShowMoreTest() {
+  it('should be active', () => {
+    isPaginationNotVisible();
+
+    backtoTopIsNotVisible();
+    scrollToFooter();
+    backToTopIsVisisble();
+  });
+
+  it('should reset the list when sorting', () => {
+    verifySortingResetsList();
+  });
+
+  it('should reset the list when filtering', () => {
+    verifyFilterResetsList();
+  });
+
+  it('should reset the list when in grid view', () => {
+    verifyGridResetsList();
+  });
+}
+
+export function infiniteScrollWithShowMoreAtTheBeginningTest() {
+  it('should be active', () => {
+    isPaginationNotVisible();
+
+    scrollToFooter(true);
+    backToTopIsVisisble();
```

Comment:
we are only asserting that the "Back to Top" button is visible when we've reached the bottom of the list (full loaded). Perhaps we can also assert at the bottom of the page, every time we scroll? So, we check the "Show More" button, and also the "Back to Top" button (but I guess only valid after the 2nd time we scroll -> maybe getting too complicated?"